Bowling/NUnitTestBowling/BowlingTests.cs

```diff
@@ -0,0 +1,70 @@
+using NUnit.Framework;
+using System.Runtime.InteropServices;
+using BowlingApp;
+using System.IO;
+using System;
+using System.Collections.Generic;
+
+namespace NUnitTestBowling
+{
+    public class Tests
+    {
+
+        [Test]
+        [TestCase("XXXXXXXXXXXX", "300", Description = "Perfect Game")]
+        [TestCase("9-9-9-9-9-9-9-9-9-9-", "90", Description = "Open Frames with -")]
+        [TestCase("5/5/5/5/5/5/5/5/5/5/5", "150", Description = "All Spares")]
+        [TestCase("548/XX725/7/52X6/5", "158", Description = "Random Open Spares Strikes")]
+        [TestCase("54637281233445625381", "82", Description = "All Open")]
+        public void TestCompleteScoringOfInputString(string marks, string expectedScore)
+        {
+            using (var sw = new StringWriter())
+            {
+                // user string for bowling record
+                using (var sr = new StringReader(marks))
+                {
+                    Console.SetOut(sw);
+                    Console.SetIn(sr);
```

Comment:
This is a cool method for controlling the input and output, kudos on figuring this out. This also makes it work well as an integration test.

Since you are doing the setup of these for all the tests in this class, I'd like to suggest doing this setup in a method adorned with the `SetUpAttribute` (see my other comment about that in this file). Note that to do so properly you'll likely have to make `sw` and `sr` instance variables and initialize them in your setup method paired with a method adorned with [TearDownAttribute](https://docs.nunit.org/articles/nunit/writing-tests/attributes/teardown.html) that disposes of them. Note that your teardown method will always be called after a test finishes even if that test throws an exception, so it's safe to put your `Dispose` calls there.